dashboard/agent.py

```diff
@@ -166,8 +168,14 @@ async def _check_parent():
 
         runner = aiohttp.web.AppRunner(app)
         await runner.setup()
-        site = aiohttp.web.TCPSite(runner, self.ip, 0)
-        await site.start()
+        try:
+            site = aiohttp.web.TCPSite(runner, self.ip, self.listen_port)
+            await site.start()
+        except OSError:
+            logger.exception("Error while trying to start HTTP Server:")
+            logger.info("Attempting to launch HTTP Server with IP: `0.0.0.0`")
+            site = aiohttp.web.TCPSite(runner, "0.0.0.0", self.listen_port)
```

Comment:
I am not sure if binding to a wildcard address is a good idea. The agent is forked from the raylet, it should be always bind to the same address with raylet. I think the `0.0.0.0` is too broad to bind.

Reply:
As we know, the raylet and dashboard agent gRPC server are listening on the wildcard address. Maybe we can listen on the wildcard address for dashboard agent http server, too.

Reply:
@fyrestone So this PR is okay then?

Reply:
Can we just `site = aiohttp.web.TCPSite(runner, "0.0.0.0", self.listen_port)` instead of try except?

Reply:
@fyrestone SGTM!!